CMakeLists.txt

```diff
@@ -84,6 +84,11 @@ endif()
 
 set(Trilinos_PREFIX "" CACHE STRING "Trilinos installation directory")
 
+option(CXX_COMPLEX "Use std::complex<double> instead of double_complex" OFF)
+if(CXX_COMPLEX)
+  add_definitions(-DCXX_COMPLEX)
```

Comment:
It looks like `apfComplexType.h` is public so the `CXX_COMPLEX` compile flag will have to be passed along.  Alternatively, `apfComplexType.h` could be defined at compile time to avoid passing the compiler flag.

If we stick with the flag, should we prefix `CXX_COMPLEX` with `APF_` or `PUMI_` to avoid collisions?

Reply:
Yeah I could just use cmake to process an apfComplexType.h.in file to create the apfComplexType.h file to avoid passing that along. I'll add a prefix as well, I think PUMI is preferable since we might wind up extending support outside of APF at some point.

Reply:
Renamed PUMI_CXX_COMPLEX, setting this value to on results in `#define PUMI_CXX_COMPLEX` being placed in `apfComplexType.h`, otherwise it is `#define PUMI_C_COMPLEX` and we don't have to carry the compile flag forward.  

See a9ebda59d55e08f54ae856b2a4fe5962f5816441